engine/src/render/camera/plugin.cpp

```diff
@@ -1,10 +1,42 @@
 #include <cubos/engine/render/camera/draws_to.hpp>
 #include <cubos/engine/render/camera/perspective_camera.hpp>
+#include <cubos/engine/render/camera/camera.hpp>
 #include <cubos/engine/render/camera/plugin.hpp>
+#include <cubos/engine/render/target/plugin.hpp>
+#include <cubos/engine/render/target/target.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+
+using namespace cubos::engine;
 
 void cubos::engine::cameraPlugin(Cubos& cubos)
 {
+    cubos.depends(renderTargetPlugin);
+
     cubos.component<PerspectiveCamera>();
+    cubos.component<Camera>();
 
     cubos.relation<DrawsTo>();
+
+    cubos.observer("add Camera on add PerspectiveCamera")
+        .onAdd<PerspectiveCamera>()
+        .without<Camera>()
+        .call([](Commands cmds, Query<Entity> query) {
+            for (auto [ent] : query)
+            {
+                cmds.add(ent, Camera{});
+            }
+        });
+
+    cubos.system("update Camera projection by PerspectiveCamera")
+        .call([](Query<Camera&, const PerspectiveCamera&, const DrawsTo&, const RenderTarget&> query) {
+            for (auto [camera, perspective, drawsTo, target] : query)
+            {
+                float aspect = static_cast<float>(target.size.x) / static_cast<float>(target.size.y);
```

Comment:
You also need to multiply target.size.x and target.size.y by the drawsTo.viewportSize!
Otherwise, when in splitscreen, the aspect ratio won't be calculated correctly.
Check out the main render sample - it's left pane becomes distorted without this.